client/src/app.js

```diff
@@ -65,10 +47,114 @@ export default class AppComponent extends HTMLElement {
+                return;
+            }
+            if (mutation.target.getAttribute("isready") === "true") {
+                this.timer.resetTimer();
+                this.timer.startTimer();
+            } else if (mutation.target.getAttribute("isready") === "false") {
+                this.timer.stopTimer();
+            }
+        }
+    }
+    
+    const observer = new MutationObserver(mutationCallback);
+    const element = this.#_shadowRoot.getElementById("playerA");
+
+    if(!element) {
+        window.setTimeout(this.listenForGame, 1000);
+
+        return;
+    }
+
+    observer.observe(this.#_shadowRoot.getElementById("playerA"), { attributes: true });
   }
+
+  sendMessageInChat = (input) => {
+    if(input.value.trim() === '') {
+        return;
+      }
+    
+      this.socketService.emitMessage(input.value);
+      input.value = '';
+  }
+
+  listenForChatMessage = () => {
+    const input = this.#_shadowRoot.getElementById('input')
+    this.#_shadowRoot.getElementById('send').addEventListener('click', () => this.sendMessageInChat(input));
+
+    input.addEventListener('keydown', (e) => {
+        if (e.key === 'Enter') {
+            this.sendMessageInChat(input);
+        }
+    })
+  }
+
+  recieveMesageFromSocket(response, isOur) {
+    const messageContainer = document.createElement('div');
+    messageContainer.classList.add('messageContainer');
+    if(!isOur) {
+        messageContainer.classList.add('left');
+    }
+
+    const innerMessage = document.createElement('div');
+    innerMessage.classList.add('message');
+    if(!isOur) {
+        innerMessage.classList.add('foreign');
+    }
+
+    const messageInfo = document.createElement('div');
+    messageInfo.classList.add('messageInfo');
+    
+    const username = document.createElement('p');
+    // TODO: Get the current user's username
+    username.innerText = 'example@gmail.com';
+    username.classList.add('username');
+    
+    const date = document.createElement('p');
+    date.innerText = moment().format('MMMM Do YYYY, h:mm:ss a'); 
+    date.classList.add('date');
+    
+    const textContainer = document.createElement('div');
+    textContainer.classList.add('textContainer');
+    
+    const textParagraph = document.createElement('p');
+    textParagraph.innerText = response.message;
+    
+    messageContainer.appendChild(innerMessage);
+    innerMessage.appendChild(messageInfo);
+    messageInfo.appendChild(username);
+    messageInfo.appendChild(date);
+    innerMessage.appendChild(textContainer);
+    textContainer.appendChild(textParagraph);
+    
+    const mainMessageContainer = this.#_shadowRoot.querySelectorAll('.message-wrapper')[0];
+    mainMessageContainer.prepend(messageContainer);
+  }
+
+  connectedCallback() {
+    this.listenForGame();
+    this.listenForChatMessage();
+  }
+
 }
 
 customElements.define('app-root', AppComponent);
 
-const app = new AppComponent();
+export const app = new AppComponent(new SocketService());
```

Comment:
As suggested in another comment, let's not do that and just use the singleton service directly in the component. Similar to how we do things in the back end